pkg/kubelet/container/helpers.go

```diff
@@ -397,3 +397,27 @@ func MakePortMappings(container *v1.Container) (ports []PortMapping) {
 	}
 	return
 }
+
+// HasAnyRegularContainerStarted returns true if any regular container has
+// started, which indicates all init containers have been initialized.
+func HasAnyRegularContainerStarted(spec *v1.PodSpec, statuses []v1.ContainerStatus) bool {
+	if len(statuses) == 0 {
+		return false
+	}
+
+	containerNames := make(map[string]struct{})
+	for _, c := range spec.Containers {
+		containerNames[c.Name] = struct{}{}
+	}
+
+	for _, status := range statuses {
+		if _, ok := containerNames[status.Name]; !ok {
+			continue
+		}
+		if status.State.Running != nil || status.State.Terminated != nil {
```

Comment:
I wonder why this is or-ed with `status.State.Terminated != nil`? If all containers have been stopped (for e.g, after node reboot), this will report that there are regular container started, which seems to be a false assertion?

Reply:
we will loose this conversation for sure. If you think it is wrong, can you please file a bug for discussion?

Reply:
https://github.com/kubernetes/kubernetes/issues/126650